test/micro/org/openjdk/bench/java/io/DataOutputStreamTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2020, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.openjdk.bench.java.io;
+
+import org.openjdk.jmh.annotations.*;
+
+import java.io.*;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(value = 1, warmups = 0)
+@Measurement(iterations = 6, time = 1)
+@Warmup(iterations=2, time = 2)
+@State(Scope.Benchmark)
+public class DataOutputStreamTest {
+
+    public enum BasicType {CHAR, SHORT, INT, STRING}
+    @Param({"CHAR", "SHORT", "INT", /* "STRING"*/}) BasicType basicType;
+
+    @Param({"4096"}) int size;
+    final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream(size);
+    final File f = new File("DataOutputStreamTest.tmp");
+    String outputString;
+    FileOutputStream fileOutputStream;
+    DataOutput bufferedFileStream, rawFileStream, byteArrayStream;
+
+    @Setup(Level.Trial)
+    public void setup() {
+        try {
+            fileOutputStream = new FileOutputStream(f);
+            byteArrayStream = new DataOutputStream(byteArrayOutputStream);
+            rawFileStream = new DataOutputStream(fileOutputStream);
+            bufferedFileStream = new DataOutputStream(new BufferedOutputStream(fileOutputStream));
+            outputString = new String(new byte[size]);
+        } catch (Exception e) {
```

Comment:
You can do `public void setup() throws Exception`, and ditch the `try-catch` block. Let JMH deal with exceptions, if any.